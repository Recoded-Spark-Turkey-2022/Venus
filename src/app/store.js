// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
