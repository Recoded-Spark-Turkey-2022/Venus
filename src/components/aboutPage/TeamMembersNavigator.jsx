import React from 'react'
import { useTranslation } from 'react-i18next';
import TeamNavButton from './TeamNavButton'

export default function TeamMembersNavigator({setter}) {
  const {t} = useTranslation()
  return (
    <div className='flex justify-center mt-[62px] mb-[82px]'>
      <TeamNavButton onClick={() => setter("View All")}>{t("About.our-team-countries.view-all")}</TeamNavButton>
      <TeamNavButton onClick={() => setter("USA")}>{t("About.our-team-countries.USA")}</TeamNavButton>
      <TeamNavButton onClick={() => setter("Turkey")}>{t("About.our-team-countries.Turkey")}</TeamNavButton>
      <TeamNavButton onClick={() => setter("Lebanon")}>{t("About.our-team-countries.Lebanon")}</TeamNavButton>
    </div>
  )
}
