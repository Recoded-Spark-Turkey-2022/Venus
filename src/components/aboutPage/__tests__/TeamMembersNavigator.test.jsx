import React from 'react';
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { store } from '../../../app/store';
import { BrowserRouter as Router } from "react-router-dom";
import TeamMembersNavigator from "../TeamMembersNavigator";

jest.mock('swiper/react', () => ({
    Swiper: 'Swiper',
    SwiperSlide: 'SwiperSlide'
  }));
  
  jest.mock('swiper', () => ({
    Pagination: 'Pagination'
  }));
  
  jest.mock('react-i18next', () => ({
    useTranslation: () => {
      return {
        t: (str) => str
      };
    },
  }));

  it(`renders correctly` , () => {
    const tree = renderer.create(
    <Router>
      <Provider store={store}>
        <TeamMembersNavigator />
        </Provider>
    </Router>);
    expect(tree).toMatchSnapshot();
});