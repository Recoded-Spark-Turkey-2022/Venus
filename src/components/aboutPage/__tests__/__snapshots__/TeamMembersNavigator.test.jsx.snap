// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex justify-center mt-[62px] mb-[82px]"
>
  <button
    className="text-[#025594] font-medium rounded-full hover:bg-[#025594] hover:text-[#ffffff] px-[17px] py-[4px]"
    onClick={[Function]}
    type="button"
  >
    About.our-team-countries.view-all
  </button>
  <button
    className="text-[#025594] font-medium rounded-full hover:bg-[#025594] hover:text-[#ffffff] px-[17px] py-[4px]"
    onClick={[Function]}
    type="button"
  >
    About.our-team-countries.USA
  </button>
  <button
    className="text-[#025594] font-medium rounded-full hover:bg-[#025594] hover:text-[#ffffff] px-[17px] py-[4px]"
    onClick={[Function]}
    type="button"
  >
    About.our-team-countries.Turkey
  </button>
  <button
    className="text-[#025594] font-medium rounded-full hover:bg-[#025594] hover:text-[#ffffff] px-[17px] py-[4px]"
    onClick={[Function]}
    type="button"
  >
    About.our-team-countries.Lebanon
  </button>
</div>
`;
